src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Composition/DependencyInjection.cs

```diff
@@ -0,0 +1,46 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Azure.Tools.GeneratorAgent.Interfaces;
+using Azure.Tools.GeneratorAgent.Services;
+using Azure.Tools.GeneratorAgent.Configuration;
+using Azure.Tools.GeneratorAgent.Logger;
+using Azure.AI.Agents.Persistent;
+using Azure.Identity;
+
+namespace Azure.Tools.GeneratorAgent.Composition
+{
+    public static class DependencyInjection
+    {
+        public static IServiceProvider Configure()
+        {
+            ServiceCollection serviceCollection = new ServiceCollection();
+
+            // Get the directory where the tool is installed
+            string toolDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) 
+                ?? throw new InvalidOperationException("Unable to determine tool installation directory");
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(toolDirectory)
+                .AddJsonFile("appsettings.json", optional: false)
+                .AddJsonFile($"appsettings.{Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development"}.json", optional: true)
```

Comment:
I wonder if we should create an environment variable name that is more specific to our context.   Maybe `AZURE_GENERATOR_ENVIRONMENT` or similar?    _(can't be AZURE_ENVIRONMENT as this has a different meaning)_

Prior art:
  - [Azure.Identity/src/EnvironmentVariables.cs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/EnvironmentVariables.cs#L10)
  - [Azure.Core.TestFramework/src/TestEnvironment.cs](https://github.com/Azure/azure-sdk-for-net/blob/012e30561f0118bf6d8df496b16c3f4d47eb256d/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L144)